src/bitwise/core.cljs

```diff
@@ -24,27 +25,26 @@
 
 (defonce game-state (r/atom (default-state)))
 (defonce dt-mult nil)
+(defonce event-chan (chan))
+(defonce event-pub (async/pub event-chan :type))
 
 (def program-catalog {:work {:name "work"
                              :complexity 1.5
                              :memory 128
-                             :on-complete (fn [state] (update-in state [:resources :data] (comp
-                                                                                           (partial into {})
-                                                                                           (partial map (fn [[key val]]
-                                                                                                          [key (inc val)])))))}})
+                             :on-complete {:type :increase-resource
```

Comment:
Providing an action to dispatch seems like a good tradeoff here, especially when a lot of programs will probably do the same thing: increment some resource by a given amount.